src/openeo_test_suite/lib/compliance_util.py

```diff
@@ -0,0 +1,532 @@
+    test_name: the name of the test that ought to be displayed
+    """
+    print("")
+    print("------------------------------------------")
+    print(
+        f'Validation Errors from path: "{endpoint_path}" Path description: {test_name}'
+    )
+
+    # message is important
+    if type(e) is UnexpectedStatusCodeException:
+        print("")
+        print(e.message)
+        if e.auth and (e.actual_status_code == 500):
+            print(
+                "Bearer-Token invalid/no longer valid or if endpoint expects an id the item does not exist."
+            )
+        elif e.actual_status_code == 500:
+            print("Endpoint expects an id and the item does not exist.")
+        elif e.actual_status_code == 404 or e.actual_status_code == 501:
+            print("Endpoint is not implemented, only an error if it is REQUIRED.")
+        elif e.actual_status_code == 410:
+            print(
+                "Endpoint is not providing requested resource as it is gone. Logs are not provided if job is queued or created."
+            )
+        else:
+            print(f"Some other unexpected error code. {e.actual_status_code}")
+    # json_path and message are important
+    elif hasattr(e.__cause__, "schema_errors"):
+        errors = e.__cause__.schema_errors
+        for error in errors:
+            print("")
+            print(error.message)
+            print(error.json_path)
+    else:
+        print(e)
+    print("------------------------------------------")
+    print("")
+
+
+def check_test_results(e: Exception):
+    """
+    prints the results of a openapi-core validation test
+
+    e: the exception that was raised as a part of the response validation test
+    test_name: the name of the test that ought to be displayed
+    """
+
+    # message is important
+
+    fail_log = ""
+    if type(e) is UnexpectedStatusCodeException:
+        if e.auth and (e.actual_status_code == 500):
+            fail_log = "Bearer-Token invalid/no longer valid or if endpoint expects an id the item does not exist."
+        elif e.actual_status_code == 500:
+            fail_log = "Endpoint expects an id and the item does not exist."
+        elif e.actual_status_code == 404 or e.actual_status_code == 501:
+            fail_log = "Endpoint is not implemented, only an error if it is REQUIRED."
+        elif e.actual_status_code == 410:
+            fail_log = "Endpoint is not providing requested resource as it is gone. Logs are not provided if job is queued or created."
+        else:
+            fail_log = f"Some other unexpected error code. {e.actual_status_code}"
+    # json_path and message are important
+    elif hasattr(e.__cause__, "schema_errors"):
+        errors = e.__cause__.schema_errors
+        for error in errors:
+            fail_log += f"Message: {error.message} Json_path: {error.json_path} \n"
+    else:
+        fail_log = str(e)
+
+    return fail_log
+
+
+def adjust_server(path_to_file, endpoint):
```

Comment:
Sure, I'll write a comment for it.

The gist is that any validation call will always fail if the url in the spec (under the server field) isn't the one the call is being sent to. This is a limitation or I guess a feature of the openapi-core library. 

openAPI also does not support having a more generic server field to reflect all the backends accurately so that this wouldn't have to be done.